webservice/s3_file_handler.py

```diff
@@ -0,0 +1,136 @@
+#! /usr/bin/env python
+
+import requests
+import boto3
+import botocore.session
+from botocore.exceptions import ClientError
+
+
+class S3FileHandler:
+    def __init__(self, region, access_key_id, secret_key):
+        """
+        This class simplifies some actions with AWS S3 storage.
+        See https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList
+        for an exhaustive list of error codes.
+
+        :param region: AWS region
+        :type region: str
+        :param access_key_id: AWS access key ID
+        :type access_key_id: str
+        :param secret_key: AWS secret access key
+        :type secret_key: str
+        """
+        service = 's3'
+        self.bucket = boto3.client(service,
+                                   aws_access_key_id=access_key_id,
+                                   aws_secret_access_key=secret_key)
+        self.resource = boto3.resource(service)
+        session = botocore.session.get_session()
+        self.session = session.create_client(service, region)
+        self.region = region
+
+    def bucket_exists(self, bucket):
+        """
+        Check whether bucket with this name exists in AWS account.
+        :param bucket: name of bucket in account
+        :type bucket: str
+        :return: HTTP status code
+        :rtype: integer
+        """
+        try:
+            self.resource.meta.client.head_bucket(Bucket=bucket)
+            return 200  # bucket exists and access authorized
+        except ClientError as e:
+            # If a client error is thrown, then check that it was a 404 error.
+            # If it was a 404 error, then the bucket does not exist.
+            error_code = int(e.response['Error']['Code'])
+            if error_code == 403:
+                return 403  # access to bucket not authorized
+            elif error_code == 404:
+                return 404  # bucket does not exist
+
+    def create_bucket(self, bucket_name):
+        """
+        :param bucket_name: name of bucket in account
+        :type bucket_name: str
+        :return: response with details
+        :rtype: JSON / Python dict
+        """
+        if self.bucket_exists(bucket_name) == 200:
+            self.bucket_name = bucket_name
+            return self.bucket.create_bucket(
+                Bucket=bucket_name,
+                CreateBucketConfiguration={'LocationConstraint': self.region}
+            )
+
+    def list_objects_in_bucket(self, bucket):
```

Comment:
I'm not sure we need this one, but it doesn't hurt.

Reply:
I left it in for now.